### ISSUE TITLE
Shipping and payment methods disappears from the list if you de-activate it for a single hub

### ISSUE THREAD

**Issue by RachL** [openfoodfoundation/openfoodnetwork#4837]
## Description

We currently have a checkbox that enables a hub to de-activate a shipping or payment method.

When a hub has only one shop, de-selecting this checkbox will make the method disappear from the list of methods for this hub. This means that they are re-creating needed shipping or payment method each time they de-activate them.

For some hub in France this means recreating a shipping method once a month....

When you have several shops, the shipping method remains active on the other shops, so still in the list.

The hub should see de-activated shipping or payments methods, even if the hub has only one shop.

## Steps to Reproduce
1. Create an account with only one hub
2. Create 2 shipping methods
3. Deactivate one of the shipping method
4. Notice you are left with only one shipping method on your list
5. Logout and log back in as a super admin, notice the shipping method is still there, just invisible for the hub.

Same for payment methods

## Animated Gif/Screenshot
Example of duplicate for hub ID 922:

## Workaround
Recreate the shipping method.

## Severity
bug-s3: a feature is broken but there is a workaround

## Your Environment
* Version used: 2.7.5
* Browser name and version: All browsers
* Operating System and version (desktop or mobile): Desktop

**Comment by gao329700254** [comment]
@RachL 
I created 2 shipping methods and deactivated one of them in an admin account with only one hub. I can still find the deactivated method in /admin/enterprises/:enterprise_name/edit#/shipping_methods. Could you please tell me which page the method disappeared? 

Env:
- Version used: 2.7.5 and 2.8.0
- Browser name and version: Chrome 80.0.3987.149
- Operating System and version (desktop or mobile): Desktop

**Comment by RachL** [comment]
@gao329700254  hello! You are on the right page but I suspect you are looking at the page with super admin rights. When you are a manager of a single enterprise, you won't see the shipping method on this page anymore.
Currently as a super admin I'm re-activating the shipping method of my single-enterprise managers each week because of this :'(

**Comment by Matt-Yorkley** [comment]
Yep, your user in development should have the admin role. You can do this:

- create a new user
- as your admin user go to Enterprises -> Edit -> Users tab, and add the new user as a manager of the enterprise you're testing
- log out and log in as the new user

You should then see that the page behaves as described

**Comment by RachL** [comment]
I've updated the issue description because I've noticed that it can happen for users with payment method shared across several shops.

**Comment by drummer83** [comment]
Just a note:
I can confirm that the bug described in steps 1 to 5 is still present in v4.1.10.
BUT: I was not able to recreate the bug for multiple hubs (steps 1 to 3). I tried with two hubs and the payment methods are still shown in the list (shown on the right), even when deactivated (shown on the left).
Or maybe I got the scenario wrong...
![Bildschirmfoto vom 2021-12-16 00-35-26](https://user-images.githubusercontent.com/1790176/146288360-c4afcdde-ff88-4655-b02f-e4497b7c5aa9.png)

Interesting to mention that the method is set to "not active" but in the lists on the right it is shown as "active". Is this a bug or is the "active" on the right more an "assigned to" and the "active" on the left is for really deactivating the method?
